cl/lib/elasticsearch_utils.py

```diff
@@ -3379,3 +3374,93 @@ def set_child_docs_and_score(
         # Optionally merges the BM25 score for display in the API.
         if merge_score and isinstance(result, AttrDict):
             result["bm25_score"] = result.meta.score
+
+
+def get_court_opinions_counts(search_query: Search) -> dict[str, int] | None:
+    """Retrieve the opinion counts per each court.
+
+    :param search_query: The ES DSL Search object.
+    :return: A dict mapping court IDs to their respective counts of
+    opinions, or None if an error occurs during query execution.
+    """
+
+    search_query = search_query.extra(size=0)
+    search_query = search_query.query(
+        Q("bool", must=Q("match", cluster_child="opinion"))
+    )
+    search_query.aggs.bucket("court_id", A("terms", field="court_id.raw"))
```

Comment:
The Elasticsearch documentation indicates that, by default, the terms aggregation returns the top ten terms with the highest document frequency. Our previous build_court_count_query method used a negative facet.limit value, which caused Solr to return an unlimited number of constraint counts. To avoid limiting the results, we should tweak this query.

Reply:
Yeah, you're absolutely right. Thanks for catching that. I've added a size parameter to this aggregation, which is the count of the Courts QuerySet. This way, the maximum size corresponds to the current number of courts in the DB, while in Elasticsearch, there might be fewer courts since not all of them may have related content indexed.